### ISSUE TITLE
Does not load on Astro View Transitions

### ISSUE THREAD

**Issue by jcf0419** [xstevenyung/astro-htmx#1]
Hey I was hoping for an easy button on loading htmx when using view transitions.. however when I navigate and have content on the destination page with hx attributes. HTMX is not picked up. Any ideas?

**Comment by netaisllc** [comment]
I experienced similar (but not using this integration. Instead I just load **htmx** via a `script` tag in my `Layout`.). 

I'm betting if you inspect the page you'll see **htmx** is loaded but, as you say, there doesn't seem to be any click handlers alive after a view transition.

FWIW, I think it's how the View Transition (VT) aspect works in Astro as opposed to this integration b/c when I remove VT from my project all works as expected.  My theory is that the implementation of the VT is blocking or hiding the page-load state from **htmx** so he doesn't know to reparse the page for any new `hx-` attributes, but that's just a guess.

**Comment by excessivecoding** [comment]
thanks @netaisllc for the additional precision. after some research, it seems like the issue is coming from either HTMX or Astro.

I found a work around by Flavio [here](https://flaviocopes.com/htmx-and-astro-view-transitions/) that seems to work would love to know if this solve your issue @jcf0419

**Comment by jcf0419** [comment]
@xstevenyung 

The following works for me.

I have the code under a single page in Astro that inherits the layout with the View Transition there.


```html

<Layout>
 <div id="fetch-jobs" hx-get="/api/rfp/getjobs" hx-swap="outerHTML" hx-trigger="load delay:200ms"></div>
</Layout>
 
 <script>
     import htmx from 'htmx.org'
     const triggerFetchLastUpdated = () => {
       const contentElement = document.getElementById('fetch-jobs')
       if (contentElement) {
         htmx.trigger(contentElement, 'click', {})
       }
     }
   
     htmx.onLoad(triggerFetchLastUpdated)
   
     document.addEventListener('astro:after-swap', () => {
       htmx.process(document.body)
       triggerFetchLastUpdated()
     })
 </script>
```

**Comment by netaisllc** [comment]
@xstevenyung 

This wrapper component seems to do it for my project.  (Not actually sure I need to declare `window` but it's true and lets the reader know what is actually going on.)

* https://gist.github.com/netaisllc/3cc72bb35ace64ceca4425374e452930

**Comment by netaisllc** [comment]
Also, I'd close this one... :)

**Comment by excessivecoding** [comment]
thanks for the solutions provided, i will close this for now

**Comment by excessivecoding** [comment]
reopenning this, i think it would make sense to include this code snippet into this package. @netaisllc i like your solution idea, would you be down to contribute to this package?

**Comment by netaisllc** [comment]
@xstevenyung Yeah sure.  But I guess I need a hint or two.  I cloned this project can't suss out how to make the integration function properly. (I don't use integrations in my Astro work.)

**Comment by excessivecoding** [comment]
in the `src/index.js` under `hooks.astro:config:setup`, you could do a separate `injectScript` which is basically the same as manually writing `<script />` tag on every astro page and paste your snippet into the second argument.

if you can i would love to the view transition into the `example`, so we can test it internally (i can also pass on the PR afterward to add the example if you don't feel confortable doing it, just up to you)

**Comment by netaisllc** [comment]
The `example` project doesn't run locally (can't load the astro config file) so why don' you just do you need to do based on my gist. That's fine with me.

**Comment by excessivecoding** [comment]
fixed with 1dd8846d05a0bfc78c5e1b049b3b03dc5790fc3b
